sqlglot/dialects/bigquery.py

```diff
@@ -247,6 +249,26 @@ def _str_to_datetime_sql(
     return self.func(f"PARSE_{dtype}", fmt, this, expression.args.get("zone"))
 
 
+def _annotate_math_functions(self: TypeAnnotator, expression: E) -> E:
+    """
+    Many BigQuery math functions such as CEIL, FLOOR etc follow this return type convention:
+    +---------+---------+---------+------------+---------+
+    |  INPUT  | INT64   | NUMERIC | BIGNUMERIC | FLOAT64 |
+    +---------+---------+---------+------------+---------+
+    |  OUTPUT | FLOAT64 | NUMERIC | BIGNUMERIC | FLOAT64 |
+    +---------+---------+---------+------------+---------+
+    """
+    self._annotate_args(expression)
+
+    this = expression.args.get("this")
+    input_type = this.type if this else exp.DataType.Type.UNKNOWN
```

Comment:
When is `this` falsey here?

Reply:
Shouldn't be, mostly satisfying the linter because these objects can be optional. Should we go with `type: ignore`?

Reply:
Maybe just `this: exp.Expression = expression.this`?